From fe62b85484dc701cf21908b23a31357d823b37cf Mon Sep 17 00:00:00 2001
From: Huicong Xu <xhc@rock-chips.com>
Date: Mon, 22 Jan 2018 09:45:30 +0800
Subject: [PATCH] phy: rockchip: inno-hdmi: fix hdmi no display sometime wake
 up

when wake up only power on hdmi phy, it may result in tmds pll unlock
because when in deep suspend, the power maybe no in a normal state.
now reinstall all register when wark up.

Change-Id: Ie882fa9b99bc6f4bfb3b2a6ea88a043b2f89ed58
Signed-off-by: Huicong Xu <xhc@rock-chips.com>
---
 drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c b/drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c
index 37477abdf521..5b28b33e8be4 100644
--- a/drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c
+++ b/drivers/phy/rockchip/phy-rockchip-inno-hdmi-phy.c
@@ -409,6 +409,7 @@ static int inno_hdmi_phy_power_off(struct phy *phy)
 	if (inno->plat_data->ops->power_off)
 		inno->plat_data->ops->power_off(inno);
 
+	inno->tmdsclock = 0;
 	dev_dbg(inno->dev, "Inno HDMI PHY Power Off\n");
 
 	return 0;
-- 
2.35.3

